### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW31)

### PR DESCRIPTION
**Pull request by alexbatashev**
LLVM: https://github.com/llvm/llvm-project/commit/5dd566b7c7b78bd385418c72d63c79895be9ae97
SPIRV-LLVM-Translator: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/c35ca633bb8bec648d23db5fa4d668b8b423da69

### CODE REVIEW

File: llvm-spirv/test/DebugInfo/X86/live-debug-variables.ll [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ; RUN: llvm-spirv -r %t.spv -o - | llvm-dis -o %t.ll
 
 ; RUN: llc -mtriple=x86_64-linux-gnu -filetype=obj -o - %t.ll | llvm-dwarfdump -debug-loc - | FileCheck %s
+; XFAIL: *
```
Comment (vladimirlaz) [inline llvm-spirv/test/DebugInfo/X86/live-debug-variables.ll:6]: You can cherry-pick change under review: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/650/files

File: llvm-spirv/test/DebugInfo/X86/xray-split-dwarf-interaction.ll [hunk 2]
```diff
@@ -11,7 +11,7 @@
 ; RUN:     -function-sections -data-sections \
 ; RUN:     -relocation-model=pic -filetype=obj \
 ; RUN:     -generate-type-units -o - %s | \
-; RUN:     llvm-readelf -sections | \
+; RUN:     llvm-readelf -sections - | \
```
Comment (vladimirlaz) [inline llvm-spirv/test/DebugInfo/X86/xray-split-dwarf-interaction.ll:14]: Can be removed like in https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/650/files

**Comment by bader** [comment]
@alexbatashev you can contact @AGindinson for resolving CI issue with FPGA test.

**Comment by bader** [comment]
If possible, could you also cherry-pick https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/93339201c7fbc158d7967b98fff156ded1a597ff to this pulldown, please?

**Comment by alexbatashev** [comment]
@bader Are you ok with disabling failing test while @AGindinson is working on a proper fix? We can then manually cherry-pick and enable it back.

**Comment by AGindinson** [comment]
> @bader Are you ok with disabling failing test while @AGindinson is working on a proper fix? We can then manually cherry-pick and enable it back.

https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/655 should fix the problem.

**Comment by bader** [comment]
> > @bader Are you ok with disabling failing test while @AGindinson is working on a proper fix? We can then manually cherry-pick and enable it back.
> 
> [KhronosGroup/SPIRV-LLVM-Translator#655](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/655) should fix the problem.

Let's try to use cherry-pick this patch. If it doesn't work, I'm okay to disable.

**Comment by AGindinson** [comment]
Does the merge commit message need to be updated, given the latest Translator additions?

**Comment by bader** [comment]
@alexbatashev, could you also fix the link to LLVM project commit in the description, please? It gives me 404.

**Comment by alexbatashev** [comment]
> Does the merge commit message need to be updated, given the latest Translator additions?

> @alexbatashev, could you also fix the link to LLVM project commit in the description, please? It gives me 404.

Done

**Pull request merged**